File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Resources/Base/MainActivity.fs
namespace ${ROOT_NAMESPACE}

open System
open System.Collections.Generic
open System.Linq
open System.Text

open Android.App
open Android.Content
open Android.OS
open Android.Runtime
open Android.Views
open Android.Widget

[<Activity (Label = "${PROJECT_NAME}")>]
type ${PROJECT_NAME}() =
  inherit Activity()

  override x.OnCreate(bundle) =
    base.OnCreate (bundle)
    // Create your application here

File: src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Resources/Base/MainActivity.cs
using System;

using Android.App;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;
//${USINGS}

namespace ${ROOT_NAMESPACE}
{
	[Register ("${JAVA_PACKAGENAME}.MainActivity"), Activity (Label = "${PROJECT_NAME}", MainLauncher = true, Icon = "@drawable/icon")]
	//${ATTRIBUTES}
	public class MainActivity : Activity
	{
		//${FIELDS}
		int count = 1;

		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);

			// Set our view from the "main" layout resource
			SetContentView (Resource.Layout.Main);

			// Get our button from the layout resource,
			// and attach an event to it
			Button button = FindViewById<Button> (Resource.Id.myButton);

			button.Click += delegate {
				button.Text = string.Format ("{0} clicks!", count++);
			};

			//${AFTER_ONCREATE}
		}
	}
	//${AFTER_MAINACTIVITY}
}


